core/trino-main/src/main/java/io/trino/operator/scalar/JsonExtract.java

```diff
@@ -118,17 +116,13 @@
 {
     private static final int ESTIMATED_JSON_OUTPUT_SIZE = 512;
 
-    private static final JsonFactory JSON_FACTORY = TrinoJsonFactory.builder()
-            .disable(CANONICALIZE_FIELD_NAMES)
-            .build();
-
     private JsonExtract() {}
 
     public static <T> T extract(Slice jsonInput, JsonExtractor<T> jsonExtractor)
     {
         requireNonNull(jsonInput, "jsonInput is null");
         try {
-            try (JsonParser jsonParser = createJsonParser(JSON_FACTORY, jsonInput)) {
+            try (JsonParser jsonParser = createJsonParser(NON_CANONICALIZING_JSON_FACTORY, jsonInput)) {
```

Comment:
it may not be obvious from refactor:

we gained `.configure(INTERN_FIELD_NAMES, false)` here now.

call it out in commit message